src/utilities/components.tsx

```diff
@@ -44,7 +44,9 @@ export function isElementOfType<P>(
     return false;
   }
 
-  const {type} = element;
+  const {type: defaultType} = element;
+  const overrideType = element.props?.__type__;
+  const type = overrideType || defaultType;
```

Comment:
Duplicating my comment on https://github.com/Shopify/remote-ui/pull/29

__type__ is very vague. The code in polaris-react isn't clear that this is due to argo-related functionality. I worry that in 3 month when people come to look at the code in polaris-react they won't have a clue what the extra logic is for and thus might be deleted.

Could this be named more explicitly? A longer name more explicit name would also reduce the potential for accidental naming clashes and thus accidental triggering of the logic in polaris-react. `__remoteUiChildType__` perhaps?

I'd also like to see a comment in here that describes what this does and why as it a very special one-off case and it is very important that in the future we understand why this exists so it doesn't get removed.

Reply:
I think we wanted to keep it decoupled from remote UI itself. Not sure that anyone would want to piggy back. Although, we're trying to move away from this pattern in the future, so it might not be the worst to be very specific. +1 on the commenting though.

Reply:
As Alan mentioned, we should eventually come up with something else. Can I keep the name and just put comment, link to the original issue instead? @BPScott

Reply:
IMO, this is actually a Polaris issue that is not isolated to usage in Argo. The following React component has the same problem as Argo's usage of Polaris does, and could be solved with the same approach:

```tsx
const MyStackItem = ({children}) => <Stack.Item fill>{children}</Stack.Item>
```

Our eventual preference is to no longer rely on the existing implementation of `Stack` or other Polaris components that have this kind of child wrapping issue, so I think it's better if, in the Polaris codebase, we describe why we need this feature more generally than just "for Argo".